esphome/components/lvgl/widgets/arc.py

```diff
@@ -58,17 +71,34 @@ def __init__(self):
         )
 
     async def to_code(self, w: Widget, config):
-        if CONF_MIN_VALUE in config:
+        if CONF_MIN_VALUE in config and CONF_MAX_VALUE in config:
             max_value = await lv_int.process(config[CONF_MAX_VALUE])
             min_value = await lv_int.process(config[CONF_MIN_VALUE])
             lv.arc_set_range(w.obj, min_value, max_value)
-            start = await lv_angle_degrees.process(config[CONF_START_ANGLE])
-            end = await lv_angle_degrees.process(config[CONF_END_ANGLE])
-            rotation = await lv_angle_degrees.process(config[CONF_ROTATION])
-            lv.arc_set_bg_angles(w.obj, start, end)
-            lv.arc_set_rotation(w.obj, rotation)
-            lv.arc_set_mode(w.obj, literal(config[CONF_MODE]))
-            lv.arc_set_change_rate(w.obj, config[CONF_CHANGE_RATE])
+        elif CONF_MIN_VALUE in config:
+            max_value = lv_expr.arc_get_max_value(w.obj)
+            min_value = await lv_int.process(config[CONF_MIN_VALUE])
+            lv.arc_set_range(w.obj, min_value, max_value)
+        elif CONF_MAX_VALUE in config:
+            max_value = await lv_int.process(config[CONF_MAX_VALUE])
+            min_value = lv_expr.arc_get_min_value(w.obj)
+            lv.arc_set_range(w.obj, min_value, max_value)
+
+        await w.set_property(
+            CONF_START_ANGLE,
+            await lv_angle_degrees.process(config.get(CONF_START_ANGLE)),
+        )
+        await w.set_property(
+            CONF_END_ANGLE, await lv_angle_degrees.process(config.get(CONF_END_ANGLE))
+        )
+        await w.set_property(
+            CONF_ROTATION, await lv_angle_degrees.process(config.get(CONF_ROTATION))
+        )
+        await w.set_property(CONF_MODE, literal(config.get(CONF_MODE)))
```

Comment:
`literal` is redundant here, `set_property()` applies that automatically to a string. In fact it can be reduced to this, as `set_property()` will also pull the value out of a dict.
The other instances above won't work like that as they need the `process()` call to convert to the right type.

```suggestion
        await w.set_property(CONF_MODE, config)
```

There's also `w.get_property()` that could be used above where you retrieve the existing max and min values. But what you have is fine too.

Reply:
You're right, I didn't spend enough time looking at set_property. Got distracted by figuring out how exactly it was generating the right functions.

I like consistency and latest/greatest, changed to get_property.